drivers/scheduler/eks/eks.go

```diff
@@ -0,0 +1,304 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/sched-ops/task"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	kube "github.com/portworx/torpedo/drivers/scheduler/k8s"
+	"github.com/portworx/torpedo/pkg/log"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	// SchedName is the name of the EKS scheduler driver implementation
+	SchedName = "eks"
+	// defaultEKSUpgradeTimeout is the default timeout for EKS control plane upgrade
+	defaultEKSUpgradeTimeout = 90 * time.Minute
+	// defaultEKSUpgradeRetryInterval is the default retry interval for EKS control plane upgrade
+	defaultEKSUpgradeRetryInterval = 5 * time.Minute
+)
+
+type EKS struct {
+	kube.K8s
+	clusterName     string
+	region          string
+	config          aws.Config
+	eksClient       *eks.Client
+	pxNodeGroupName string
+}
+
+// String returns the string name of this driver.
+func (e *EKS) String() string {
+	return SchedName
+}
+
+func (e *EKS) Init(schedOpts scheduler.InitOptions) (err error) {
+	// This implementation assumes the EKS cluster has two node groups: one group for
+	// Torpedo and another group for Portworx.
```

Comment:
Don't think this comment belongs here anymore

Reply:
I thought so too, but I believe that by default, we deploy EKS clusters with Torpedo in a separate node group from Portworx, so I left a comment about it in the Init section.

Reply:
But I will move it to UpgradeScheduler to avoid confusion. Thanks.